tst/EnergyPlus/unit/OutputProcessor.unit.cc

```diff
@@ -5350,6 +5352,138 @@ namespace OutputProcessor {
         for (int i = 0; i < (int)reportDataDictionary.size(); ++i)
             EXPECT_EQ(reportDataDictionary[i], reportDataDictionaryResults[i]);
     }
+
+    TEST_F(EnergyPlusFixture, OutputProcessor_StdoutRecordCount)
+    {
+        auto &op = state->dataOutputProcessor;
+        std::string const idf_objects = delimited_string({
+            "Output:Meter,Electricity:Facility,Timestep;",
+            "Output:Variable,*,Lights Electricity Energy,Timestep;",
+        });
+
+        ASSERT_TRUE(process_idf(idf_objects));
+
+        // Demonstrate that it's not unreasonable to reach the INT_MAX limit at all
+        constexpr int numOfTimeStepInHour = 60;
+        constexpr int hoursInYear = 8760;
+        constexpr int numOutputVariables = 4200;
+        constexpr size_t totalRecords =
+            static_cast<size_t>(numOfTimeStepInHour) * static_cast<size_t>(hoursInYear) * static_cast<size_t>(numOutputVariables);
+
+        constexpr auto INT_MAX_AS_SIZE_T = static_cast<size_t>(INT_MAX);
+        EXPECT_GT(totalRecords, INT_MAX_AS_SIZE_T);
```

Comment:
It's not unreasonable to hit the INT_MAX limit

Reply:
OK OK I said I believed you.